platform/jobs/edxPlatformLettuceMaster.groovy

```diff
@@ -0,0 +1,149 @@
+package devops
+
+import hudson.model.Build
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_WORKER
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_PARSE_SECRET
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_HIPCHAT
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_BASE_URL
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_JUNIT_REPORTS
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_GITHUB_STATUS_SUCCESS
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_GITHUB_STATUS_UNSTABLE_OR_WORSE
+
+/*
+Example secret YAML file used by this script
+publicJobConfig:
+    open : true/false
+    jobName : name-of-jenkins-job-to-be
+    testengUrl: github-url-segment
+    platformUrl : github-url-segment-clone
+    testengCredential : n/a
+    platformCredential : n/a
+    platformCloneReference : clone/.git
+    email : email-address@email.com
+    hipchat : token
+*/
+
+Map <String, String> predefinedPropsMap  = [:]
+predefinedPropsMap.put('GIT_SHA', '${GIT_COMMIT}')
+predefinedPropsMap.put('GITHUB_ORG', 'edx')
+predefinedPropsMap.put('GITHUB_REPO', 'edx-platform')
+predefinedPropsMap.put('TARGET_URL', JENKINS_PUBLIC_BASE_URL + 'job/edx-platform-lettuce-master/${BUILD_NUMBER}/')
+predefinedPropsMap.put('CONTEXT', 'jenkins/lettuce')
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+/* TODO: Move this into JenkinsPublicConstants, as it can be shared. */
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+stringParam = [
```

Comment:
nit: in the other jobs (e.g. edxPlatformJsMaster, edxPlatformQualityMaster), this variable is named "params". Might be better to keep it for consistency. Also the code gets confusing to read later on when you then need to say: `stringParam(stringParam.name, stringParam.default, stringParam.description)`